test/integration/http2_integration_test.cc

```diff
@@ -1245,6 +1245,34 @@ TEST_P(Http2IntegrationTest, PauseAndResumeHeadersOnly) {
   ASSERT_TRUE(response->complete());
 }
 
+TEST_P(Http2IntegrationTest, ContinueHeadersOnlyInjectBodyFilter) {
```

Comment:
Can you put this in protocol_integration_test? It would be good to get coverage of all protocol permutations.